components/CharmCalculator.tsx

```diff
@@ -0,0 +1,198 @@
+"use client";
+
+import { useState } from "react";
+
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from "@/components/ui/select";
+import { Table, TableBody, TableCaption, TableCell, TableHead, TableHeader, TableRow } from "@/components/ui/table";
+
+import Heroes from "../public/data/heroes.json";
```

Comment:
merging this into the existing `lib/characters.ts` would be preferred.

also the `@` in an import statement refers to the root of the project. relative imports are very fragile.